### PR TITLE
[wasm64] Fix reading/writing of gl attributes

### PR DESCRIPTION
**Pull request by sbc100**
See #21177

### CODE REVIEW

File: src/library_html5_webgl.js [hunk 1]
```diff
@@ -30,12 +30,12 @@ var LibraryHtml5WebGL = {
       HEAP32[a+i] = 0;
     }
 
-    HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.alpha }}}>>2)] =
```
Comment (dschuff) [inline src/library_html5_webgl.js:33]: I assume the size of EM_BOOL hasn't changed, was this just wrong before?

File: test/float_tex.cpp [hunk 2]
```diff
@@ -9,10 +9,9 @@
 #include <cmath>
 #include <iostream>
 #include <vector>
-extern "C" {
```
Comment (dschuff) [inline test/float_tex.cpp:12]: A lot of these changes look unrelated?

File: src/library_html5_webgl.js [hunk 3]
```diff
@@ -80,24 +80,23 @@ var LibraryHtml5WebGL = {
 #if ASSERTIONS
     assert(attributes);
 #endif
-    var a = attributes >> 2;
-    var powerPreference = HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.powerPreference }}}>>2)];
```
Comment (dschuff) [inline src/library_html5_webgl.js:84]: This set of changes look like just changing one form of i32 memory reference to another; where's the 64-bit fix?

File: test/float_tex.cpp (hunk 2 above)
Comment (sbc100) [reply test/float_tex.cpp:12]: Yeah, I just cleaned up the test a little while I was working on it.

File: src/library_html5_webgl.js (hunk 3 above)
Comment (sbc100) [reply src/library_html5_webgl.js:84]: The problem is that >>2 doesn't work for numbers bigger and 2**32.

**Comment by sbc100** [comment]
Sorry I had another change mixed up in here.. fixed now.   Looking into avoiding the size regression somehow.

File: src/library_html5_webgl.js [hunk 4]
```diff
@@ -30,12 +30,12 @@ var LibraryHtml5WebGL = {
...
-    HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.antialias }}}>>2)] =
-    HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.premultipliedAlpha }}}>>2)] =
-    HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.majorVersion }}}>>2)] =
-    HEAP32[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.enableExtensionsByDefault }}}>>2)] = 1;
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.alpha }}})] =
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.depth }}})] =
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.antialias }}})] =
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.premultipliedAlpha }}})] =
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.majorVersion }}})] =
+    HEAP8[a + ({{{ C_STRUCTS.EmscriptenWebGLContextAttributes.enableExtensionsByDefault }}})] = 1;
```
Comment (sbc100) [inline src/library_html5_webgl.js:38]: This is just way to set the low bit of each of these 32-bit values.  We set the whole struct to 1 above.. then set these individual fields to 1.

Its a bit of a hack but it avoid the shift (which is the problem.

File: src/library_html5_webgl.js [hunk 5]
```diff
@@ -30,12 +30,12 @@ var LibraryHtml5WebGL = {
       HEAP32[a+i] = 0;
```
Comment (dschuff) [inline src/library_html5_webgl.js:30]: do the references on lines 18 and 29 above also need to be fixed?

File: src/library_html5_webgl.js (hunk 4 above)
Comment (dschuff) [reply src/library_html5_webgl.js:38]: Ah I see. This seems kind of fragile though, like it could easily get switched back. maybe it should at least have a comment.

File: src/library_html5_webgl.js (hunk 1 above)
Comment (sbc100) [reply src/library_html5_webgl.js:33]: This change is no longer part of this PR now that #21189 has landed

File: test/float_tex.cpp [hunk 6]
```diff
@@ -79,7 +80,9 @@ static void updateFloatTexture() {
     glBindTexture(GL_TEXTURE_2D, 0);
     alpha -= 0.001;
 }
+
 static void glut_draw_callback(void) {
+    printf("glut_draw_callback\n");
```
Comment (dschuff) [inline test/float_tex.cpp:85]: do we want to keep these printfs in the test?

Comment (sbc100) [reply test/float_tex.cpp:85]: Sure, I think it better overall to have more print statements in the test.

**Comment by sbc100** [comment]
Closing in favor of #21225 21228
